### PR TITLE
Release NPM package, and Node update

### LINKED ISSUE #679: Create npm package for Node.js sdk

### PR DESCRIPTION
**Pull request by markmandel**
Made sense to do these two parts together, since they were intertwined.

1. Closed #679 - create an interactive shell, so we can publish to @google-cloud/agones-sdk, since it requires some internal tools to make it happen.
2. Also update the Node version to 12.x, since 11.x is deprecated
3. Updated the image to use the new NPM package, and also while in there updated Node to 12.x and the Debian base to Buster. (Node image update for #1261)
4. Updated the release checklist to use the interactive shell, rather than the `publish` command, which has now been removed.

### CODE REVIEW

File: site/content/en/docs/Guides/Client SDKs/nodejs.md [hunk 1]
```diff
@@ -22,9 +18,12 @@ Download the source {{< ghlink href="sdks/nodejs" >}}directly from GitHub{{< /gh
 Add the agones dependency to your project:
 
 ```sh
-$ npm install @googleforgames/agones
+$ npm install @google-cloud/agones-sdk
```
Comment (markmandel) [inline site/content/en/docs/Guides/Client SDKs/nodejs.md]: Didn't feature shortcode wrap this, as to test, I pushed up a 1.3.0 npm image, so it works as of right now.

File: build/build-sdk-images/node/Dockerfile [hunk 2]
```diff
@@ -14,7 +14,7 @@
 ARG BASE_IMAGE=agones-build-sdk-base:latest
 FROM $BASE_IMAGE
 
-RUN curl -sL https://deb.nodesource.com/setup_11.x | bash - && \
+RUN curl -sL https://deb.nodesource.com/setup_12.x | bash - && \
```
Comment (steven-supersolid) [inline build/build-sdk-images/node/Dockerfile]: Oh I didn't spot this before but generally best to stick with the LTS version which will always be even and is currently 12. So just for future reference so we don't upgrade to 13

File: build/includes/sdk.mk [hunk 3]
```diff
@@ -170,6 +170,17 @@ run-sdk-conformance-tests: run-sdk-conformance-test-node run-sdk-conformance-tes
 clean-sdk-conformance-tests:
 	$(MAKE) run-all-sdk-command COMMAND=clean
 
+# Start a shell in the SDK image. This is primarily used for publishing packages.
+# Usig a shell is the easiest, because of Google internal processes and interactive commands required.
```
Comment (steven-supersolid) [inline build/includes/sdk.mk]: nit: Typo Usig -> Using

File: build/includes/sdk.mk [hunk 4]
```diff
@@ -170,6 +170,17 @@ run-sdk-conformance-tests: run-sdk-conformance-test-node run-sdk-conformance-tes
...
+sdk-shell:
+	$(MAKE) ensure-build-sdk-image SDK_FOLDER=$(SDK_FOLDER)
+	docker run --rm -it $(common_mounts) -v ~/.ssh:/tmp/.ssh:ro $(DOCKER_RUN_ARGS) \
+ 	--entrypoint=/root/shell.sh $(SDK_IMAGE_TAG)
+
+# 1SDK shell for node
```
Comment (steven-supersolid) [inline build/includes/sdk.mk]: Is the 1 a typo or some acronym?

Comment (markmandel) [reply build/includes/sdk.mk]: Typo, will fix! Thanks for catching that!

**Pull request merged**